### PR TITLE
session_value

### PR DESCRIPTION
**Pull request by thejacer87**

### CODE REVIEW

File: src/Tags/Base.php [hunk 1]
```diff
@@ -264,4 +264,24 @@ protected function hiddenFieldForSelect($options) {
             [ 'type' => 'hidden', 'value' => '']
         ));
     }
+
+    protected function getOldInput() {
+        $index = $this->objectName;
```
Comment (aksonnic) [inline src/Tags/Base.php]: Is this ok for simply nested attrs like `person.address.province`? Won't `objectname` in that case just be `address`?

File: tests/views/flash/nested_checkbox.blade.php [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{{ \Form::with(null, ['url'=>'/test-other-route']) }}
+  {{ \Form::checkBoxWithObject('post[some][nested][path]', 'is_published', $options) }}
+{{ \Form::end() }}
```
Comment (aksonnic) [inline tests/views/flash/nested_checkbox.blade.php]: This is not nested from Forms point of view, but not a bad test by itself.

Add a test with `Form::fieldsFor` nesting

File: src/Tags/Base.php [hunk 3]
```diff
@@ -267,13 +267,14 @@ protected function hiddenFieldForSelect($options) {
...
-            $name = $this->getNameFromOptions($this->options['index'], $this->options);
-            $temp = preg_replace('/\]\[|[^-a-zA-Z0-9_\-:.]/', '.', $name);
-            $index = preg_replace('/\.$/', '', $temp);
+            $index = $this->getNameFromOptions($this->options['index'], $this->options);
+            $temp = preg_replace('/\]\[|[^-a-zA-Z0-9_\-:.]/', '.', $index);
+            $index_dotted = preg_replace('/\.$/', '', $temp);
         }
 
-        $old_input = request()->old($index);
+        $old_input = request()->old($index) ?? request()->old($index . "[$this->attr]") ?? request()->old($index_dotted) ?? request()->old($index_dotted . ".$this->attr");
```
Comment (aksonnic) [inline src/Tags/Base.php]: Why the heck do we need to check allllll those different ways? Everything in `old()` should be dotted, no?

Comment (thejacer87) [reply src/Tags/Base.php]: updated

File: src/Tags/Base.php [hunk 4]
```diff
@@ -267,19 +267,22 @@ protected function hiddenFieldForSelect($options) {
...
         if (isset($this->options['index'])) {
             $index = $this->getNameFromOptions($this->options['index'], $this->options);
-            $temp = preg_replace('/\]\[|[^-a-zA-Z0-9_\-:.]/', '.', $index);
-            $index_dotted = preg_replace('/\.$/', '', $temp);
         }
+        $temp = preg_replace('/\]\[|[^-a-zA-Z0-9_\-:.]/', '.', $index);
+        $index = preg_replace('/\.$/', '', $temp);
 
-        $old_input = request()->old($index) ?? request()->old($index . "[$this->attr]") ?? request()->old($index_dotted) ?? request()->old($index_dotted . ".$this->attr");
+        $old_input = request()->old($index) ?? request()->old($index . ".$this->attr");
```
Comment (aksonnic) [inline src/Tags/Base.php]: Why the `??`. `old($index)` is never what you want, is it?

**Pull request merged**